index.js

```diff
@@ -1,55 +1,371 @@
+    cartSpan.innerHTML = `${cartItemToRender.quantity}`
+    makeCartLi.append(cartSpan)
+
+    // make, edit, append, listen - BUTTON ADD
+    const cartAddButton = document.createElement('button')
+    cartAddButton.setAttribute('class', 'quantity-btn add-btn center')
+    cartAddButton.innerText = '+'
+    cartAddButton.addEventListener('click', () => {
+      // increase quantity
+      cartItemToRender.quantity++
+      renderCart(cartItemToRender)
+    })
+    makeCartLi.append(cartAddButton)
+}
+console.log('*** UPDATED STATE CART ***', stateCart)
+cartTotal()
+}
+
+// * RENDER TOTAL
+
+// calculate cart total from item quantities and prices
+function cartTotal() {
+  console.log('called: cartTotal')
+  let currentTotal = 0
+  for (let i = 0; i < stateCart.length; i++) {
+    let currentItemTotal
+    currentPrice = stateCart[i].price
+    currentQuantity = stateCart[i].quantity
+    currentItemTotal = currentQuantity * currentPrice
+    currentTotal += currentItemTotal
+     // format for currency
+    currentTotalFormatted = Intl.NumberFormat('en-UK').format(currentTotal) 
+  // render the total on the page
+    findTotal = document.querySelector('.total-number')
+    findTotal.innerHTML = ''
+    findTotal.innerText = `£${currentTotalFormatted}`
+  }
+  console.log('__________________')
+}
+
+// * RENDER FILTERS
+
+// display filter buttons for - all, fruit and veg
+function renderFilters() {
+  console.log('called: renderFilters')
+
+  // ALL div
+  const allFilter = document.createElement('div')
+  allFilter.setAttribute('class', 'all-filter filter')
+  selectFilters.insertBefore(allFilter, selectStoreUl)
+  // all filter button
+  const allFilterButton = document.createElement('button')
+  allFilterButton.innerText = 'Show All'
+  allFilter.appendChild(allFilterButton)
+  // listen for fruit click
+  allFilterButton.addEventListener('click', () => {
+    console.log('fruit filter clicked')
+    state.showFruit = true
+    state.showVeg = true
+    renderStore()
+  })
+
+  // FRUIT div
+  const fruitFilter = document.createElement('div')
+  fruitFilter.setAttribute('class', 'fruit-filter filter')
+  selectFilters.insertBefore(fruitFilter, selectStoreUl)
+  // fruit filter button
+  const fruitFilterButton = document.createElement('button')
+  fruitFilterButton.innerText = 'Show only fruit'
+  fruitFilter.appendChild(fruitFilterButton)
+  // listen for fruit click
+  fruitFilterButton.addEventListener('click', () => {
+    console.log('fruit filter clicked')
+    state.showFruit = true
+    state.showVeg = false
+    renderStore()
+  })
+
+  // VEG div
+  const vegFilter = document.createElement('div')
+  vegFilter.setAttribute('class', 'veg-filter filter')
+  selectFilters.insertBefore(vegFilter, selectStoreUl)
+  // veg filter button
+  const vegFilterButton = document.createElement('button')
+  vegFilterButton.innerText = 'Show only veg'
+  vegFilter.appendChild(vegFilterButton)
```

Comment:
The buttons become hidden outside of the screen when I scroll down, maybe try a different way of positioning them?

Reply:
Good shout, I was just focussed on functionality - can definitely go back in and fix the visuals for it